src/settings/ViewPrintDetails/ViewPrintDetails.js

```diff
@@ -0,0 +1,126 @@
+  const {
+    printDetailsInfo,
+    enablePrintLog,
+    isLoading,
+    refetch,
+  } = usePrintDetailsSettings();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [reRenderFlag, setReRenderFlag] = useState(true);
+  const { createPrintDetailsSettings, updatePrintDetailsSettings } = usePrintDetailsSettingsMutation();
+  const { formatMessage } = useIntl();
+  const callout = useCallout();
+
+  const createSetting = (values) => {
+    return createPrintDetailsSettings({
+      name: PRINT_DETAILS_SETTINGS_KEY,
+      value: {
+        enablePrintLog: values[VIEW_PRINT_DETAILS_ENABLED].toString(),
+      }
+    });
+  };
+
+  const updateSetting = (values) => {
+    return updatePrintDetailsSettings({
+      ...printDetailsInfo,
+      value: {
+        ...printDetailsInfo.value,
+        enablePrintLog: values[VIEW_PRINT_DETAILS_ENABLED].toString(),
+      }
+    });
+  };
+
+  const apiHandler = (values) => {
+    const handler = printDetailsInfo?.id ? updateSetting : createSetting;
+
+    return handler(values)
+      .then(() => {
+        refetch();
+        callout.sendCallout({
+          type: 'success',
+          message: formatMessage(
+            { id: 'ui-circulation.settings.ViewPrintDetails.callout.success' },
+          ),
+        });
+      })
+      .catch(() => {
+        callout.sendCallout({
+          type: 'error',
+          message: formatMessage(
+            { id: 'ui-circulation.settings.ViewPrintDetails.callout.error' },
+          ),
+        });
+      });
+  };
+
+  const onSubmit = async (values) => {
+    if (!values[VIEW_PRINT_DETAILS_ENABLED]) {
+      setIsModalOpen(true);
+    } else {
+      apiHandler(values);
+    }
+  };
+
+  const handleModalClose = () => {
+    setReRenderFlag(!reRenderFlag);
+    setIsModalOpen(false);
+  };
+
+  const handleModalYes = () => {
+    apiHandler({ [VIEW_PRINT_DETAILS_ENABLED]: false });
+    setIsModalOpen(false);
+  };
+
+  const modalFooter = (
+    <ModalFooter>
+      <Button
+        onClick={handleModalYes}
+        buttonStyle="primary"
+      >
+        {formatMessage({ id: 'ui-circulation.settings.ViewPrintDetails.warningPopupMessage.yes' })}
+      </Button>
+      <Button
+        onClick={handleModalClose}
+        buttonStyle="primary"
+      >
+        {formatMessage({ id: 'ui-circulation.settings.ViewPrintDetails.warningPopupMessage.no' })}
+      </Button>
+    </ModalFooter>
+  );
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  return (
+    <>
+      <ViewPrintDetailsForm
+        onSubmit={onSubmit}
+        key={reRenderFlag}
+        initialValues={{ [VIEW_PRINT_DETAILS_ENABLED]: enablePrintLog }}
+      />
+      <Modal
+        data-testid="viewPrintDetailsModal"
+        open={isModalOpen}
+        onClose={handleModalClose}
+        dismissible
+        footer={modalFooter}
+      >
+        {formatMessage({ id: 'ui-circulation.settings.ViewPrintDetails.warningPopupMessage' })}
+      </Modal>
```

Comment:
1. This Modal doesn't have a title. I do not either see it on ticket description. Could you please check across folio apps or check with Kimie and Irina, if this is acceptable? https://github.com/folio-org/stripes-components/blob/32ab3daaa0d773e0c27b447f205d092cf8561b48/lib/Modal/Modal.js#L46 states that modal title or label is a required field. But is missing here.
2. Implementation with <Modal> component is fine, but usually such functionality is implemented using https://github.com/folio-org/stripes-components/tree/master/lib/ConfirmationModal and this